Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/ACL/OnUpdate.hs

```diff
@@ -69,18 +72,26 @@ data OnUpdateBuildReq
       }
 
 buildOnUpdateMessage ::
-  (EsqDBFlow m r, EncFlow m r) =>
+  (EsqDBFlow m r, EncFlow m r, EsqDBReplicaFlow m r) =>
   OnUpdateBuildReq ->
   m OnUpdate.OnUpdateMessage
 buildOnUpdateMessage RideAssignedBuildReq {..} = do
   mobileNumber <- SP.getPersonNumber driver >>= fromMaybeM (InternalError "Driver mobile number is not present.")
   name <- SP.getPersonFullName driver >>= fromMaybeM (PersonFieldNotPresent "firstName")
+  inProgressRide <- Ride.getInProgressByDriverId driver.id
+  newRide <- Ride.getNewByDriverId driver.id
+  driverLastDropLocationLocation <- case inProgressRide of
+    Just _ -> case newRide of
+      Just _ -> Esq.runInReplica (SQB.findByDriverIdTripEndLatLonIfRideStatusNew driver.id)
+      Nothing -> pure Nothing
+    Nothing -> pure Nothing
   let agent =
         RideAssignedOU.Agent
           { name = name,
             phone = mobileNumber,
             rating = realToFrac <$> driver.rating,
-            tags = RideAssignedOU.AgentTags {registered_at = driver.createdAt}
+            tags = RideAssignedOU.AgentTags {registered_at = driver.createdAt},
+            driverLastDropLocationLocation = driverLastDropLocationLocation
```

Comment:
@NikithShetty  @hkmangla is this fine here or it should be in `AgentTags` or somewhere else i.e. `tags` inside `StartInfo` ?